packages/account/src/Sections/Profile/PersonalDetails/__tests__/verify-button.spec.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { VerifyButton } from '../verify-button';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { routes } from '@deriv/shared';
+import { usePhoneNumberVerificationSetTimer } from '@deriv/hooks';
+
+jest.mock('@deriv/hooks', () => ({
+    ...jest.requireActual('@deriv/hooks'),
+    usePhoneNumberVerificationSetTimer: jest.fn(),
+}));
+
+describe('VerifyButton', () => {
+    beforeEach(() => {
+        (usePhoneNumberVerificationSetTimer as jest.Mock).mockReturnValue({ next_otp_request: '' });
+    });
+    const history = createBrowserHistory();
+    const renderWithRouter = (component: React.ReactElement) => {
+        return render(<Router history={history}>{component}</Router>);
+    };
+
+    const mock_store = mockStore({
+        client: {
+            account_settings: {
+                phone_number_verification: {
+                    verified: 0,
+                },
+            },
+        },
+    });
+
+    it('should render Verify Button', () => {
+        renderWithRouter(
+            <StoreProvider store={mock_store}>
```

Comment:
I see everywhere we use StoreProvider with VerifyButton, let's also move this code to renderWithRouter func